client/python/conductor/FrinxConductorWrapper.py

```diff
@@ -24,140 +29,196 @@
+
+            registerd_task: RegisteredWorkerTask = self.task_types[task_type]
+
+            self.filtered_queue[task_type] -= 1
+            if self.filtered_queue[task_type] <= 0:
+                self.filtered_queue.pop(task_type, None)
+
+            next_worker = NextWorkerTask()
+            next_worker.task_type = task_type
+            next_worker.exec_function = registerd_task.exec_function
+            next_worker.domain = registerd_task.domain
+            next_worker.poll_uuid = self.actual_uuid
+            return next_worker
+
+    def task_not_found_anymore(self, task_not_found):
+        if self.actual_uuid == task_not_found.poll_uuid:
+            with self.lock:
+                self.filtered_queue.pop(task_not_found.task_type, None)
+
+
+class NextWorkerTask:
+    def __init__(self):
+        self.task_type = None
+        self.domain = None
+        self.exec_function = None
+        self.poll_uuid = None
+
+
+
+class ExceptionHandlingConductorWrapper(ConductorWorker):
+    def __init__(self, server_url, thread_count, polling_interval, worker_id=None, headers=None, consumer_worker_count=120):
         # Synchronizes access to self.queues by producer thread (in read_queue) and consumer threads (in tasks_in_queue)
         self.lock = threading.Lock()
         self.queues = {}
         self.conductor_task_url = server_url + "/metadata/taskdefs"
-        self.headers= headers
-        super().__init__(server_url, thread_count, polling_interval, worker_id, headers)
-
-    def start_queue_polling(self):
-        log.debug("Starting queue polling thread")
-        thread = Thread(target=self.read_queues)
-        thread.daemon = True
-        thread.start()
-
-    def read_queues(self):
-        failCount = 0
-        log.debug("Queue polling thread started")
+        self.headers = headers
+        self.consumer_worker_count = consumer_worker_count
+        self.task_source = TaskSource()
+        super().__init__(server_url, thread_count, polling_interval, worker_id, self.headers)
+
+    def start_workers(self):
+        for i in range(self.consumer_worker_count):
+            thread = Thread(target=self.consume_task)
+            thread.daemon = True
+            thread.start()
+
+        logger.info("Starting a queue polling")
+        fail_count = 0
         while True:
             try:
-                time.sleep(float(self.polling_interval))
-                self.lock.acquire()
-                queuesTemp = self.taskClient.getTasksInQueue("all")
-                log.debug('Queues polled: %s', queuesTemp)
-                self.queues = queuesTemp
-                failCount = 0
-            except Exception as err:
-                log.exception('Unable to read queue info. Error count: %s', failCount, err)
-                self.queues = {}
-                failCount =+ 1
-                if (failCount > 10):
-                    log.exception('Exiting, unable to read queue info')
+                time.sleep(0.1)
```

Comment:
Shouldn't this be ?

                time.sleep(float(self.polling_interval))

Reply:
Yes, should be. Code was updated